aspnet/fundamentals/routing.rst

```diff
@@ -1,10 +1,272 @@
+
+.. image:: /fundamentals/routing/_static/default-mvc-routetemplate.png
+
+Tokens within curly braces (``{ }``) define `route value` placeholders which will be bound if the route is matched. You can define more than one route value placeholder in a route segment, but they must be separated by a literal value. For example "{controller=Home}{action=Index}" would not be a valid route, since there is no literal value between ``{controller}`` and ``{action}``. These route value placeholders must have a name, and may have additional attributes specified.
+
+The following table demonstrates the available route value placeholder options and `route value constraints <https://github.com/aspnet/Routing/blob/1.0.0-rc1/src/Microsoft.AspNet.Routing/RouteOptions.cs>`_.
+
+.. list-table:: Route Template Values
+	:header-rows: 1
+
+	* - Route Template
+	  - Example Matching URL
+	  - Notes
+	* - hello
+	  - /hello
+	  - Will only match the single path '/hello'
+	* - {Page=Home}
+	  - /
+	  - Will match and set ``Page`` to ``Home``.
+	* - {Page=Home}
+	  - /Contact
+	  - Will match and set ``Page`` to ``Contact``
+	* - {controller}/{action}/{id?}
+	  - /Products/List
+	  - Will map to ``Products`` controller and ``List`` method; Since ``id`` was not supplied in the URL, it's ignored.
+	* - {controller}/{action}/{id?}
+	  - /Products/Details/123
+	  - Will map to ``Products`` controller and ``List`` method, with ``id`` set to ``123``.
+	* - {controller=Home}/{action=Index}/{id?}
+	  - /
+	  - Will map to ``Home`` controller and ``Index`` method; ``id`` is ignored.
+
+Adding a colon ``:`` after the name allows additional constraints to be set on a route value placeholder.
+
+.. list-table:: Route Constraints
+	:header-rows: 1
+
+	* - constraint
+	  - Example
+	  - Example Match
+	  - Notes
+	* - ``int``
+	  - {id:int}
+	  - 123
+	  - Matches any integer
+	* - ``bool``
+	  - {active:bool}
+	  - true
+	  - Matches ``true`` or ``false``
+	* - ``datetime``
+	  - {dob:datetime}
+	  - 2016-01-01
+	  - Matches a valid ``DateTime`` value
+	* - ``decimal``
+	  - {price:decimal}
+	  - 49.99
+	  - Matches a valid ``decimal`` value
+	* - ``decimal``
+	  - {price:double}
+	  - 4.234
+	  - Matches a valid ``double`` value
+	* - ``float``
+	  - {price:float}
+	  - 3.14
+	  - Matches a valid ``float`` value
+	* - ``guid``
+	  - {id:guid}
+	  - 7342570B-44E7-471C-A267-947DD2A35BF9
+	  - Matches a valid ``Guid`` value
+	* - ``long``
+	  - {ticks:long}
+	  - 123456789
+	  - Matches a valid ``long`` value
+	* - ``minlength(value)``
+	  - {username:minlength(5)}
+	  - steve
+	  - String must be at least 5 characters long.
+	* - ``maxlength(value)``
+	  - {filename:maxlength(8)}
+	  - somefile
+	  - String must be no more than 8 characters long.
+	* - ``length(min,max)``
+	  - {filename:length(4,16)}
+	  - Somefile.txt
+	  - String must be at least 8 and no more than 16 characters long.
+	* - ``min(value)``
+	  - {age:min(18)}
```

Comment:
Constraints like min/max/range operate on `long` FYI. We have to convert it to a numeric type first before we can apply this logic.